ivy/functional/frontends/torch/miscellaneous_ops.py

```diff
@@ -410,3 +410,14 @@ def clone(input):
 @to_ivy_arrays_and_back
 def cov(input, /, *, correction=1, fweights=None, aweights=None):
     return ivy.cov(input, ddof=correction, fweights=fweights, aweights=aweights)
+
+
+@to_ivy_arrays_and_back
+def view_as_real(input):
+    if not ivy.is_complex_dtype(input):

```

Comment:
Is this the right way to do this? I saw that the pytorch implementation didn't accept real values. 
Also, how does the testing code test for this?

Reply:
@satakshigarg I also found that `torch.frontend` doesn't check for complex dtype in `torch.imag` as such differs in behavior from  original pytorch function implementation as shown below.
```python
import ivy
import ivy.functional.frontends.torch as t_f
import torch
# reproducable example:

a = torch.tensor([1])
b = t_f.imag(a)
print(b) # ivy.frontends.torch.Tensor([0])

a_ = torch.tensor([1])

b_ = torch.imag(a_)
print(b_) # RuntimeError: imag is not implemented for tensors with non-complex dtypes.
```

Reply:
I have also added the `with_supported_dtype` wrapper as suggested in [discord](https://discord.com/channels/799879767196958751/998782045494976522/1132013960019845192)

Reply:
> @satakshigarg I also found that `torch.frontend` doesn't check for complex dtype in `torch.imag` as such differs in behavior from  original pytorch function implementation as shown below.
> ```python
> import ivy
> import ivy.functional.frontends.torch as t_f
> import torch
> # reproducable example:
> 
> a = torch.tensor([1])
> b = t_f.imag(a)
> print(b) # ivy.frontends.torch.Tensor([0])
> 
> a_ = torch.tensor([1])
> 
> b_ = torch.imag(a_)
> print(b_) # RuntimeError: imag is not implemented for tensors with non-complex dtypes.
> ```


@satakshigarg can u please comment on this?